### ISSUE TITLE
Explore solutions for codecov upload failures

### ISSUE THREAD

**Issue by RMeli** [MDAnalysis/mdanalysis#4576]
As reported in https://github.com/MDAnalysis/mdanalysis/issues/4574, `codecov` often fails to upload reports from a fork. PR https://github.com/MDAnalysis/mdanalysis/pull/4575 removed the CI failure for `codecov` uploads, which is not an ideal long-term solution.

This issue is to track possible long-term solutions. As stated in https://github.com/MDAnalysis/mdanalysis/issues/4574#issuecomment-2068030426:

> Longer term - if nothing happens within codecov by end of May, then we should take the time to review the pro/cons of alternate providers and see which is best to move from.

**Comment by RMeli** [comment]
I was about to open another issue for the annoying `codecov` failures, then I remembered I opened this one a long while ago. Looks like `codecov` has been more annoying than useful for use (I use it in other smaller project and was quite happy with it). Should we look for alternative solutions? Should we temporarily disable the `codecov/project` check? (The last 5 pages of PRs have been merged with CI failing because of this issue.)

**Comment by IAlibay** [comment]
@RMeli have you checked that the issue here isn't just hat we haven't updated to v5 of codecov? IIRC they were meant to fix the fork upload in Q3-4 of last year, so it's very possible that v5 dealt with that.

**Comment by RMeli** [comment]
I have not. I'll check the release notes, thank you for pointing it out. Should we try to update the action?

**Comment by IAlibay** [comment]
> update the action

Yeah we should, I think it got stalled by that dependabot PR that tried to update other things that were broken.

**Comment by RMeli** [comment]
Ok, I'll give it a try when I have a bit of time (end of the week or next week)!
